crates/dht/src/dht/snapshot.rs

```diff
@@ -250,8 +250,6 @@ impl DhtSnapshot {
                 },
             ) => {
                 if our_disc_boundary != other_disc_boundary {
-                    // TODO Don't expect the boundary to move during a comparison so treat this as
```

Comment:
No longer relevant, we're permitting boundary updates during gossip rounds so this can happen and returning `CannotCompare` is the right action to take